packages/workspace/src/browser/quick-open-workspace.ts

```diff
@@ -62,6 +68,7 @@ export class WorkspaceQuickOpenItem extends QuickOpenItem {
         private readonly workspaceService: WorkspaceService,
         private readonly messageService: MessageService,
         private readonly workspace: string,
+        private readonly home: string | undefined
```

Comment:
How about precomputing `label` and `description` and passing them here. The super constructor already accepts them.

Reply:
Just to clarify, you're suggesting to remove the subclass `WorkspaceQuickOpenItem` and directly use a precomputed `label`, `description` and `run` function when creating **items** for the `QuickOpenWorkspace`?